### PR TITLE
[JENKINS-43370] Use proxy configuration from Jenkins

### PR DESCRIPTION
**Pull request by tisoft**
[JENKINS-43370](issues.jenkins-ci.org/browse/JENKINS-43370)

### CODE REVIEW

**Comment by raul-arabaolaza** [comment]
Hello,

Thanks for this PR, and sorry for the delay, I do not have much laptop access this days but I will do my best to review this as quick as possible.

File: src/main/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubStatusNotificationStep.java [hunk 1]
```diff
@@ -216,6 +219,24 @@ public String getTargetUrl() {
...
+    /**
+     * Uses proxy if configured on pluginManager/advanced page
+     *
+     * @param host GitHub's hostname to build proxy to
+     *
+     * @return proxy to use it in connector. Should not be null as it can lead to unexpected behaviour
+     */
+    @Nonnull
+    private static Proxy getProxy(@Nonnull String host) {
+        Jenkins jenkins = GitHubWebHook.getJenkinsInstance();
```
Comment (raul-arabaolaza) [inline src/main/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubStatusNotificationStep.java]: I am not sure if this is the best way to access the jenkins instance... I understand that the only reason why you introduced `GitHubWebHook` dependency was to get access to the running jenkins instance... Is there any reason for this or is just a way you found to access the jenkins instance? 

If there is no special reason I would recommend  to use `Jenkins.getInstance()`, this way we do not introduce dependencies  and in the end [GitHubWebHook.getJenkinsInstance](https://github.com/jenkinsci/github-plugin/blob/93d40692ff3866705175624e93ec584d4ac88132/src/main/java/com/cloudbees/jenkins/GitHubWebHook.java#L147) is just a little wrapper over `Jenkins.getInstance()`

**Comment by raul-arabaolaza** [comment]
@tisoft First review done, I believe there is a minor adjustment to do

**Comment by tisoft** [comment]
I have removed the `GitHubWebHook` and replaced it with `Jenkins.getInstance()`.

**Comment by raul-arabaolaza** [comment]
Hi @tisoft 

Unfortunately your changes broke the tests so we can not merge yet, if you are willing to adapt the tests it would be great if not I can take your changes and continue where you left. In any case I plan to mention you and keep your commits.

**Comment by tisoft** [comment]
I'll have a look at the tests.

**Comment by raul-arabaolaza** [comment]
Great, sorry for all the problems

**Comment by tisoft** [comment]
I have changed the tests to mock the GitHubBuilder instead of the Github class directly. The tests are running through for me now.

**Comment by raul-arabaolaza** [comment]
@tisoft Test are also running for me and maual testing is also working (see the check), given that I have to replace the current tests with ones using wiremock I am not going to get very paranoid on test style

I am going to cut a new release this afternoon, Thanks @tisoft

**Comment by stowns** [comment]
Is there a way to disable usage of the proxy config from Jenkins? We have an internal git server and while we use the jenkins proxy for calls to the outside world I do not want this plugin to use the proxy

**Comment by tisoft** [comment]
If you add your internal domains to the no Proxy Settings in Jenkins, the plugin will honor that setting.

**Comment by hadv** [comment]
I still have below issue with version 1.0.3; do you have any suggestion? Thank you!

```
java.lang.IllegalArgumentException: The suplied credentials are invalid to login
	at org.jenkinsci.plugins.pipeline.githubstatusnotification.GitHubStatusNotificationStep.getGitHubIfValid(GitHubStatusNotificationStep.java:263)
	at org.jenkinsci.plugins.pipeline.githubstatusnotification.GitHubStatusNotificationStep.getRepoIfValid(GitHubStatusNotificationStep.java:268)
	at org.jenkinsci.plugins.pipeline.githubstatusnotification.GitHubStatusNotificationStep.access$100(GitHubStatusNotificationStep.java:79)
	at org.jenkinsci.plugins.pipeline.githubstatusnotification.GitHubStatusNotificationStep$Execution.run(GitHubStatusNotificationStep.java:373)
	at org.jenkinsci.plugins.pipeline.githubstatusnotification.GitHubStatusNotificationStep$Execution.run(GitHubStatusNotificationStep.java:355)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE
```

**Comment by GearoidCollins** [comment]
Im having an issue here at this line: https://github.com/jenkinsci/pipeline-githubnotify-step-plugin/blob/master/src/main/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubStatusNotificationStep.java#L255

I am providing the `gitApiUrl` to our enterprise github account but because the proxy is not enabled on Jenkins it is not working and I am getting the error ```"The supplied credentials are invalid to login"```.

I installed `1.0.2` of this plugin and it works beautifully when specifying `gitApiUrl`.

I just want to know is there a reason for this or is it a bug? 

If there is a reason for this can we have a more intuitive exception?

**Pull request merged**